### PR TITLE
deprecate "debug" flag for rp_install

### PR DESCRIPTION
**Pull request by andre-merzky**
As the title says, this deprecates the `debug` mode for the pilot side radical stack installation in favor of the ``local` mode, which now is confined to the pilot sandbox.  This is a cleaner fix than #1166 to avoid race conditions on `pip install` during pilot startup.

### CODE REVIEW

**Comment by andre-merzky** [comment]
Note that merge is not urgent, as a simple workaround for #1166 exists, by setting `rp_version = "debug"`.

**Comment by ibethune** [comment]
Issue #1166 implies that the workaround above was implemented, so can this PR be closed?

**Comment by andre-merzky** [comment]
It should be merged I think.  Despite the large number of changed lines, its actually a rather simple fix, as it *removes* some code complexity, and from a fairly rarely used code path at that.  I propose that Iself-review that PR once more and merge toward the release, if that's ok?

**Comment by ibethune** [comment]
OK, let's go with that.

File: src/radical/pilot/bootstrapper/bootstrap_1.sh [hunk 1]
```diff
@@ -429,8 +429,8 @@ virtenv_setup()
     virtenv_mode="$3"
     python_dist="$4"
 
-    virtenv_create=UNDEFINED
-    virtenv_update=UNDEFINED
+    ve_create=UNDEFINED
+    ve_update=UNDEFINED
```
Comment (andre-merzky) [inline src/radical/pilot/bootstrapper/bootstrap_1.sh]: This name change removes an ambiguity between function and variable names.  The shell parsing had no problem with that - human parsers did... ;)

**Comment by marksantcroos** [comment]
> It should be merged I think. Despite the large number of changed lines, its actually a rather simple fix, as it removes some code complexity, and from a fairly rarely used code path at that.

And removed functionality, so I would like to have noted that I objected to this change :-)

**Comment by andre-merzky** [comment]
Interesting, I though we discussed this!  What functionality do you consider removed?  

The main difference is that the `rp_install` tree now always lives in the pilot sandbox, and never next to the global sandbox -- but the stack you can configure to get installed in `rp_install` did not change.  In `debug`, that did not give any performance boost either, as the stack was always reinstalled (which is exactly the problem this tries to solve, as pip-locking for that reinstall caused trouble).  What am I missing?

**Comment by marksantcroos** [comment]
`debug` and `local` had different semantics, but discussing them is impossible now as my head can't wrap around the renaming anymore. Let it be.

**Comment by andre-merzky** [comment]
Ok.  Once you stumble over something which you need in this context, let me know...

**Pull request merged**